### ISSUE TITLE
FeatureRequest: nested property matcher

### ISSUE THREAD

**Issue by servel333** [sinonjs/sinon#1556]
Request to add this function signature: `sinon.match.hasNested(keypath, expected)`

where `keypath` functions like [Lodash.get](https://lodash.com/docs/#get) supporting `"outer.inner"` or `"outer[inner]"`.

Turning this syntax into this syntax

    sinon.match.has('outer', sinon.match.has('middle', sinon.match.has('inner', sinon.match.string)));
    sinon.match.hasNested('outer.middle.inner', sinon.match.string);

Personally, I would prefer `has` supported nested keypaths, but that would probably break backwards compatibility. The implementation is simple if the addition of the `lodash.get` library is acceptable.

```
const get = require("lodash.get");

match.hasNested = createPropertyMatcher(function (actual, property) {
    return get(actual, property) !== undefined;
}, "hasNested");

```

**Comment by fatso83** [comment]
`lodash.get` is about 6k (2k gzipped), but we are not required to load in microseconds, so that should be fine. If you can supply a PR, that would be welcome.

**Comment by mroderick** [comment]
This was fixed with #1560
